lib/features/email/data/network/email_api.dart

```diff
@@ -402,6 +405,86 @@ class EmailAPI with HandleSetErrorMixin {
     return bytesDownloaded;
   }
 
+  Future<void> downloadAllAttachmentsForWeb(
+    AccountId accountId,
+    EmailId emailId,
+    String baseDownloadAllUrl,
+    String outputFileName,
+    AccountRequest accountRequest,
+    DownloadTaskId taskId,
+    StreamController<Either<Failure, Success>> onReceiveController,
+    {CancelToken? cancelToken}
+  ) async {
+    final authentication = accountRequest.authenticationType == AuthenticationType.oidc
+        ? accountRequest.bearerToken
+        : accountRequest.basicAuth;
+    final headerParam = _dioClient.getHeaders();
+    headerParam[HttpHeaders.authorizationHeader] = authentication;
+    headerParam[HttpHeaders.acceptHeader] = DioClient.jmapHeader;
+    
+    final downloadAllUriTemplate = UriTemplate(Uri.decodeFull(baseDownloadAllUrl));
+    final downloadAllUrl = downloadAllUriTemplate.expand({
+      'accountId': accountId.asString,
+      'emailId': emailId.asString,
+      'name': outputFileName,
+    });
+
+    final bytesDownloaded = await _dioClient.get(
+      downloadAllUrl,
+      options: Options(
+        headers: headerParam,
+        responseType: ResponseType.bytes),
+      onReceiveProgress: (downloaded, total) {
+        log('EmailAPI::downloadFileForWeb(): downloaded = $downloaded | total: $total');
+        double progress = 0;
+        if (downloaded > 0 && total >= downloaded) {
+          progress = (downloaded / total) * 100;
+        }
+        log('EmailAPI::downloadFileForWeb(): progress = ${progress.round()}%');
+        onReceiveController.add(Right(DownloadingAllAttachmentsForWeb(
+          taskId,
+          '$outputFileName.zip',
```

Comment:
Should create a variable to reuse.

```suggestion
final downloadFileName = '$outputFileName.zip';
```

Reply:
Done